docs/modules/ROOT/pages/reference/exoscale-osbapi.adoc

```diff
@@ -0,0 +1,368 @@
+
+=== Plan Change
+
+We don't support plan changes on the Exoscale console, everything is configured on our portal.
+There is only one plan per service, the default plan.
+
+One exception is the "suspension-plan" which is described in the <<Suspension Flow>>.
+
+=== Deprovisioning
+
+See <<Offboarding Flow>> as this is mostly the same flow.
+
+== Billing
+
+The basic flow: We send billing data to Exoscale, Exoscale invoices the end-user, VSHN sends an invoice to Exoscale, Exoscale pays VSHN.
+
+[mermaid]
+....
+flowchart TB
+    exo["Exoscale"]
+    exocust["Exoscale Customer"]
+    vshn["VSHN"]
+
+    exo-- Invoices --> exocust
+    exocust-- Pays -->exo
+
+    vshn-- Invoices --> exo
+    exo-- Pays -->vshn
+....
+
+Exoscale must keep track on our pricing on their end, because we only send usage data and they do the calculation.
+
+TODO
+
+* Send billing data to Exoscale billing API - Exoscale does invoicing to customer - we send invoice to Exoscale
+* One SO on VSHN side for Exoscale, to send invoice to Exoscale
+** We track the Exoscale organization ID in the SO
+** Maybe different product in product DB? Or different variant?
+* How to send billing data to Exoscale? Once per month directly from Odoo data, so that we send the same data?
+
+From Exoscale docs:
+
+____
+
+You can define one or more plans corresponding to various service offerings or service levels on your platform.
+
+*Monthly fees*
+
+Each plan can have an optional monthly fee.
+When a subscriber unsubscribes from your service, the service is cancelled immediately and they are charged with a pro-rated amount dating from their last subscription charge.
+
+*Additional charges*
+
+It is possible to charge for additional products and services in addition to the optional monthly fee.
+All additional billing dimensions must be declared in advance with a defined price for each available plan.
+
+Billing dimensions are specified by:
+
+* a technical name
+* a unit
+
+Supported units are:
+
+* h : hours
+* gb : gigabytes
+* gb.h : gigabytes per hour
+* u : arbitrary quantity
+
+The frequency of metering reporting is up to the vendor. You can meter as frequently as every hour.
+Metering should be reported at least once a month per customer.
+Metering is reported per client organization with the consumption that has occurred since the last successful report. Multiple charges can be reported at once.
+When reporting usage, you send the quantity for each defined variable and the client is charged accordingly.
+____
+
+
+[source,json]
+----
+POST /orgs/:uuid/usage <1>
+
+{
+    "records": [
+        {
+            "variable": "something",
+            "quantity": 12.5
+        },
+        {
+            "variable": "something_else",
+            "quantity": 1.2
+        }
+    ]
+}
+----
+<1> `:uuid` is the technical ID of the client organization in the Exoscale backend, which will be shared during the onboarding process.
+
+== OSBServiceInstance
```

Comment:
Wouldn't it make more sense to just track the state of the org with this CRD?

The users can't provision multiple instances via the Exoscale GUI anyway. So a `OSBOrgInstance` is probably enough?

Or is the idea that this is not Exoscale specific? Then I'd go with another name and maybe move it to the `Portal` page. As we've established, Exoscale might be the only one who actually uses OSB, so the OSB specifics could be dropped.